napari/layers/_shapes_layer/shape_list.py

```diff
@@ -0,0 +1,616 @@
+                                                vertices, axis=0)
+        vertices = shape._edge_offsets
+        self._mesh_vertices_offsets = np.append(self._mesh_vertices_offsets,
+                                                vertices, axis=0)
+        index = np.repeat([[shape_index, 1]], len(vertices), axis=0)
+        self._mesh_vertices_index = np.append(self._mesh_vertices_index, index,
+                                              axis=0)
+
+        triangles = shape._edge_triangles + m
+        self._mesh_triangles = np.append(self._mesh_triangles, triangles,
+                                         axis=0)
+        index = np.repeat([[shape_index, 1]], len(triangles), axis=0)
+        self._mesh_triangles_index = np.append(self._mesh_triangles_index,
+                                               index, axis=0)
+        color = shape.edge_color.rgba
+        color[3] = color[3]*shape.opacity
+        color_array = np.repeat([color], len(triangles), axis=0)
+        self._mesh_triangles_colors = np.append(self._mesh_triangles_colors,
+                                                color_array, axis=0)
+
+        # Add faces to mesh
+        m = len(self._mesh_vertices)
+        vertices = shape._face_vertices
+        self._mesh_vertices = np.append(self._mesh_vertices, vertices, axis=0)
+        vertices = shape._face_vertices
+        self._mesh_vertices_centers = np.append(self._mesh_vertices_centers,
+                                                vertices, axis=0)
+        vertices = np.zeros(shape._face_vertices.shape)
+        self._mesh_vertices_offsets = np.append(self._mesh_vertices_offsets,
+                                                vertices, axis=0)
+        index = np.repeat([[shape_index, 0]], len(vertices), axis=0)
+        self._mesh_vertices_index = np.append(self._mesh_vertices_index, index,
+                                              axis=0)
+
+        triangles = shape._face_triangles + m
+        self._mesh_triangles = np.append(self._mesh_triangles, triangles,
+                                         axis=0)
+        index = np.repeat([[shape_index, 0]], len(triangles), axis=0)
+        self._mesh_triangles_index = np.append(self._mesh_triangles_index,
+                                               index, axis=0)
+        color = shape.face_color.rgba
+        color[3] = color[3]*shape.opacity
+        color_array = np.repeat([color], len(triangles), axis=0)
+        self._mesh_triangles_colors = np.append(self._mesh_triangles_colors,
+                                                color_array, axis=0)
+
+        # Set z_order
+        self._update_z_order()
+
+    def remove_all(self):
+        """Removes all shapes
+        """
+        self.shapes = []
+        self._vertices = np.empty((0, 2))
+        self._index = np.empty((0), dtype=int)
+        self._z_index = np.empty((0), dtype=int)
+        self._z_order = np.empty((0), dtype=int)
+
+        self._mesh_vertices = np.empty((0, 2))
```

Comment:
If you had a mesh object you could have a MeshObject.clear() method that abstracts all this.

Reply:
I have added a `Mesh` object and a `clear` method